astropy/table/column.py

```diff
@@ -1091,28 +1131,48 @@ def insert(self, obj, values, mask=None):
 
         out = new_ma.view(self.__class__)
         out.parent_table = None
+        out.indices = []
         out._copy_attrs(self)
 
         return out
 
-    def __getitem__(self, item):
-        out = super(MaskedColumn, self).__getitem__(item)
-
+    def copy_attribs(self, out):
```

Comment:
This doesn't seem like it should be a public method.  Maybe also call it `_copy_attrs_slice` or something to emphasize its primary context.